### PR TITLE
Add admin UI for editing user roles

### PR DESCRIPTION
**Pull request by ktbee**
@evmiguel @gnarf @rwaldron here is a PR for adding an admin UI for editing user roles. 

My proposed changes:
- Add account admin components and related Redux action, reducer
- Fixes login form bug where username is required (it shouldn't be, right? Like if someone just wants to use their email address? Let me know if I have that wrong)
- Adds a SQL view for getting all users and aggregating their roles (currently each user's role is a separate row in `user_role`
- Adjust server roles routes to use the req.body and req.session for wherever we need user information and not route params.

Screen Shot 2019-11-08 at 5 38 31 PM

Let me know what you think! Currently it auto saves with no messaging, which I know users didn't love before. However I'm wondering if TM admins would be more comfortable with auto-saving.

### CODE REVIEW

File: client/components/AccountAdmin/UserRows.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    researcher: 'Researcher',
+    facilitator: 'Facilitator',
+    participant: 'Participant'
+});
+
+const onCheckboxClick = async (event, { email, username, role, checked }) => {
+    const endpoint = checked ? '/api/roles/add' : '/api/roles/delete';
+    const body = JSON.stringify({
+        roles: [role],
+        email,
```
Comment (rwaldron) [inline client/components/AccountAdmin/UserRows.jsx]: What happens if there is no value for `email`? `email` is not required for creating an account.

File: client/components/CreateAccount/index.jsx [hunk 2]
```diff
@@ -89,7 +89,6 @@ class CreateAccount extends Component {
                 <Form.Field>
                     <label htmlFor="name">Username</label>
                     <input
-                        required
```
Comment (rwaldron) [inline client/components/CreateAccount/index.jsx]: This one _is_ required. Only email is optional.

File: client/components/AccountAdmin/UserRows.jsx (hunk 1 above)
Comment (ktbee) [reply client/components/AccountAdmin/UserRows.jsx]: Currently nothing happens, it is just undefined in this req body and instead we'll use the username to get the right user to edit their roles. I have fixed/re-added the "required" attribute on the username field for account creation, so there will at least always be a username.

**Comment by ktbee** [comment]
Thanks for the clarification @rwaldron! I have re-made the username required, if you'd like to take another look.

**Pull request merged**